src/cpp.cc

```diff
@@ -1088,6 +1089,16 @@ void Printer::Cpp::print(const Statement::Table_Decl &t) {
   print(ns);
   stream << indent() << dtype << " zero;" << endl;
 
+  if (checkpoint) {
+    stream << indent() << "boost::filesystem::path out_table_path;" << endl;
+    stream << indent() << "boost::filesystem::path tmp_out_table_path;" << endl;
+    stream << indent() << "boost::filesystem::path in_table_path;" << endl;
+    stream << indent() << "std::mutex m;" << endl;
+    stream << indent() << "std::string formatted_interval;" << endl;
```

Comment:
It's kind of redundant, but you can't access any members from `out` from any of the table classes since they are part of `out` and you can't access non-static members "upwards" (if that makes sense), so I chose to simply pass `formatted_interval` to every table class in `init` and live with this redundancy.